dom/events/Event-dispatch-click.html

```diff
@@ -241,6 +241,54 @@
   input.dispatchEvent(new MouseEvent("click"));
 }, `disabled radio should fire onclick`);
 
+async_test(t => {
+  const input = document.createElement("input");
+  input.type = "checkbox";
+  input.disabled = true;
+  input.onclick = t.step_func(ev => {
+    assert_true(input.checked);
+    ev.preventDefault();
+    setTimeout(t.step_func_done(() => {
+      assert_false(input.checked);
+    }));
+  });
+  input.dispatchEvent(new MouseEvent("click", { cancelable: true }));
+}, `disabled checkbox should get legacy-canceled-activation behavior`);
+
+async_test(t => {
+  const input = document.createElement("input");
+  input.type = "radio";
+  input.disabled = true;
+  input.onclick = t.step_func(ev => {
+    assert_true(input.checked);
+    ev.preventDefault();
+    setTimeout(t.step_func_done(() => {
+      assert_false(input.checked);
+    }));
+  });
+  input.dispatchEvent(new MouseEvent("click", { cancelable: true }));
+}, `disabled radio should get legacy-canceled-activation behavior`);
+
+async_test(t => {
+  const input = document.createElement("input");
+  input.type = "checkbox";
+  input.disabled = true;
+  const ev = new MouseEvent("click", { cancelable: true });
+  ev.preventDefault();
+  input.dispatchEvent(ev);
+  setTimeout(t.step_func_done(() => assert_false(input.checked)));
+}, `disabled checkbox should get legacy-canceled-activation behavior 2`);
```

Comment:
This fails on Firefox but passes when `input.onclick = () => {}` is added. The spec does not require such thing, right?

Reply:
Right. Nice find.